readthedocs/search/tasks.py

```diff
@@ -142,19 +142,45 @@ def delete_old_search_queries_from_db():
 
 
 @app.task(queue='web')
-def record_search_query(project_slug, version_slug, query, total_results):
-    """Record search query in database."""
-    if not project_slug or not version_slug or not query or not total_results:
+def record_search_query(project_slug, version_slug, query, total_results, time):
+    """Record/update search query in database."""
+    if not project_slug or not version_slug or not query:
         log.debug(
             'Not recording the search query. Passed arguments: '
-            'project_slug: %s, version_slug: %s, query: %s, total_results: %s' % (
-                project_slug, version_slug, query, total_results
+            'project_slug: %s, version_slug: %s, query: %s, total_results: %s, time: %s' % (
+                project_slug, version_slug, query, total_results, time
             )
         )
         return
 
-    project_qs = Project.objects.filter(slug=project_slug)
+    before_10_sec = time - timezone.timedelta(seconds=10)
+    partial_query_qs = SearchQuery.objects.filter(
+        project__slug=project_slug,
+        version__slug=version_slug,
+        query__startswith=query,
+        created__gte=before_10_sec,
+    ).order_by('-created')
+
+    # if partial query exists,
+    # just update it instead of creating new SearchQuery object.
+    if partial_query_qs.exists():
+        obj = partial_query_qs.first()
```

Comment:
By doing `partial_query_qs.exists()` and then `partial_query_qs.first()` you do make 2 SQL queries. You can probably just do `partial_query_qs.first()` and check if it is `None`.